data-prepper-plugins/kinesis-source/src/main/java/org/opensearch/dataprepper/plugins/kinesis/source/processor/KinesisRecordProcessor.java

```diff
@@ -161,16 +168,21 @@ public void processRecords(ProcessRecordsInput processRecordsInput) {
 
             // Track the records for checkpoint purpose
             kinesisCheckpointerTracker.addRecordForCheckpoint(extendedSequenceNumber, processRecordsInput.checkpointer());
-            List<Record<Event>> records = kinesisRecordConverter.convert(
+
+            List<KinesisInputOutputRecord> kinesisOutputRecords = kinesisRecordConverter.convert(
                     kinesisStreamConfig.getCompression().getDecompressionEngine(),
                     processRecordsInput.records(), streamIdentifier.streamName());
 
             int eventCount = 0;
-            for (Record<Event> record: records) {
-                Event event = record.getData();
+            for (KinesisInputOutputRecord kinesisInputOutputRecord: kinesisOutputRecords) {
+                Record<Event> dataPrepperRecord = kinesisInputOutputRecord.getDataPrepperRecord();
+                long incomingRecordSizeBytes = kinesisInputOutputRecord.getIncomingRecordSizeBytes();
+                bytesReceivedSummary.record(incomingRecordSizeBytes);
+                Event event = dataPrepperRecord.getData();
                 acknowledgementSetOpt.ifPresent(acknowledgementSet -> acknowledgementSet.add(event));
 
-                bufferAccumulator.add(record);
+                bufferAccumulator.add(dataPrepperRecord);
+                bytesProcessedSummary.record(incomingRecordSizeBytes);
```

Comment:
you have this line repeated in line number 180 as well? Not sure if this is intentional